types/evidence.go

```
@@ -83,6 +83,50 @@ const (
 	MaxEvidenceBytesDenominator = 10
 )
 
+func EvidenceToProto(evidence Evidence) (*tmproto.Evidence, error) {
+	switch evi := evidence.(type) {
+	case *DuplicateVoteEvidence:
+		vote_b, err := evi.VoteB.ToProto()
+		if err != nil {
+			return nil, err
+		}
+		vote_a, err := evi.VoteA.ToProto()
+		if err != nil {
+			return nil, err
+		}
+		return &tmproto.Evidence{
+			Sum: &tmproto.Evidence_DuplicateVoteEvidence{
+				DuplicateVoteEvidence: &tmproto.DuplicateVoteEvidence{
+					VoteA: vote_a,
+					VoteB: vote_b,
+				},
+			},
+		}, nil
+	default:
+		panic("evidence is not recognized")
```

shouldn't we return an error? I think @erikgrinaker pointed this out